design/src/rfcs/rfc0041_file_per_change_changelog.md

```diff
@@ -0,0 +1,125 @@
+since the central `CHANGELOG.next.toml` file is almost always a merge conflict for two PRs
+with changelog entries.
+
+This RFC proposes a new approach to change logging that will remedy the merge conflict issue,
+and explains how this can be done without disrupting the current release process.
+
+The proposed developer experience
+---------------------------------
+
+There will be a `changelog/` directory in the smithy-rs root where
+developers can add changelog entry Markdown files. Any file name can be picked
+for these entries. Suggestions are the development branch name for the
+change, or the PR number.
+
+The changelog entry format will change to make it easier to duplicate entries
+across both smithy-rs and aws-sdk-rust, a common use-case.
+
+This new format will make use of Markdown front matter in the YAML format.
+This change in format has a couple benefits:
+- It's easier to write change entries in Markdown than in a TOML string.
+- There's no way to escape special characters (such as quotes) in a TOML string,
+  so the text that can be a part of the message will be expanded.
+
+While it would be preferable to use TOML for the front matter (and there are libraries
+that support that), it will use YAML so that GitHub's Markdown renderer will recognize it.
+
+A changelog entry file will look as follows:
+
+```markdown
+---
+# Adding `aws-sdk-rust` here duplicates this entry into the SDK changelog.
+applies_to: ["client", "server", "aws-sdk-rust"]
+authors: ["author1", "author2"]
+references: ["smithy-rs#1234", "aws-sdk-rust#1234"]
+# The previous `meta` section is broken up into its constituents:
+breaking: false
+# This replaces "tada":
+new_feature: false
+bug_fix: false
+---
+
+Some message for the change.
+```
+
+Implementation
+--------------
+
+When a release is performed, the release script will generate the release notes,
+update the `CHANGELOG.md` file, copy SDK changelog entries into the SDK,
+and delete all the files in `changelog/`.
+
+### SDK Entries
+
+The SDK changelog entries currently end up in `aws/SDK_CHANGELOG.next.json`, and each entry
+is given `age` and `since_commit` entries. The age is a number that starts at zero, and gets
+incremented with every smithy-rs release. When it reaches a hardcoded threshold, that entry
+is removed from `aws/SDK_CHANGELOG.next.json`. The SDK release process uses the `since_commit`
+to determine which changelog entries go into the next SDK release's changelog.
+
+The SDK release process doesn't write back to smithy-rs, and history has shown that it
+can't since this leads to all sorts of release issues as PRs get merged into smithy-rs
+while the release is in progress. Thus, this `age`/`since_commit` dichotomy needs to
+stay in place.
+
+The `aws/SDK_CHANGELOG.next.json` will stay in place in its current format without changes.
+Its JSON format is capable of escaping characters in the message string, so it will be
+compatible with the transition from Markdown to TOML.
```

Comment:
Is `the transition from Markdown to TOML.` meant to be `the transition from Markdown to YAML.`?

Reply:
Fixed in 0d712a2908df7c3b629b91019594737d352410c0